CMakeLists.txt

```diff
@@ -319,6 +319,13 @@ if(CMAKE_VERSION VERSION_LESS "3.7.0")
   set(CMAKE_INCLUDE_CURRENT_DIR ON)
 endif()
 
+# Export compile-commands.json and symlink it into the source directory (needed
+# for clang-tidy pre-commit hook)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+file(CREATE_LINK
+  ${CMAKE_BINARY_DIR}/compile_commands.json
+  ${CMAKE_SOURCE_DIR}/compile_commands.json SYMBOLIC)
```

Comment:
I struggled with this for hours, believe me. Supposedly adding the `-p` argument to clang-tidy allows you to specify a directory where the json resides, but it did not work for me at all. And even if it worked - how do you know the name of the cmake build directory? It can be different for everyone.
Furthermore, this also enables you to run clang-tidy manually without any setup.
And it's not like we have a super-tidy working directory now ^^